src/customFileParser/FibonacciDecoder.ts

```diff
@@ -0,0 +1,59 @@
+const FIBONACCI_MAX_LENGTH = 77; // Any 53 bit integer will take at most 77 bits in a fibonacci encoding.
+
+const fibonacciLookup = [1, 2];
+for (let i = 2; i <= FIBONACCI_MAX_LENGTH; i++) {
+  fibonacciLookup.push(fibonacciLookup[i - 1] + fibonacciLookup[i - 2]);
+}
+
+export default class FibonacciDecoder {
+  public numberRead: number;
+  private buffer: ArrayBuffer;
+  private data: Uint8Array;
+  private readBitId: number;
+  private currentValue: number;
+  private nextFibIndex: number;
```

Comment:
(also the usages of the variable)
```suggestion
  private nextIndex: number;
```

Reply:
What about this one?